nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ParseSyslog5424.java

```diff
@@ -0,0 +1,174 @@
+    @WritesAttribute(attribute = "syslog.timestamp", description = "The timestamp of the Syslog message."),
+    @WritesAttribute(attribute = "syslog.hostname", description = "The hostname or IP address of the Syslog message."),
+    @WritesAttribute(attribute = "syslog.appname", description = "The appname of the Syslog message."),
+    @WritesAttribute(attribute = "syslog.procid", description = "The procid of the Syslog message."),
+    @WritesAttribute(attribute = "syslog.messageid", description = "The messageid the Syslog message."),
+    @WritesAttribute(attribute = "syslog.structuredData", description = "Multiple entries per structuredData of the Syslog message."),
+    @WritesAttribute(attribute = "syslog.sender", description = "The hostname of the Syslog server that sent the message."),
+    @WritesAttribute(attribute = "syslog.body", description = "The body of the Syslog message, everything after the hostname.")})
+@SeeAlso({ListenSyslog.class, ParseSyslog.class, PutSyslog.class})
+public class ParseSyslog5424 extends AbstractProcessor {
+
+    public static final AllowableValue OMIT = new AllowableValue(NilPolicy.OMIT.name(),NilPolicy.OMIT.name(),"The missing field will not have an attribute added.");
+    public static final AllowableValue NULL = new AllowableValue(NilPolicy.NULL.name(),NilPolicy.NULL.name(),"The missing field will have an empty attribute added.");
+    public static final AllowableValue DASH = new AllowableValue(NilPolicy.DASH.name(),NilPolicy.DASH.name(),"The missing field will have an attribute added with the value of '-'.");
+
+    public static final PropertyDescriptor CHARSET = new PropertyDescriptor.Builder()
+        .name("Character Set")
+        .description("Specifies which character set of the Syslog messages")
+        .required(true)
+        .defaultValue("UTF-8")
+        .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+        .build();
+
+    public static final PropertyDescriptor NIL_POLICY = new PropertyDescriptor.Builder()
+        .name("nil_policy")
+        .displayName("NIL Policy")
+        .description("Defines how NIL values are handled for header fields.")
+        .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+        .allowableValues(OMIT,NULL,DASH)
+        .required(true)
+        .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+        .defaultValue(NULL.getValue())
+        .build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+        .name("failure")
+        .description("Any FlowFile that could not be parsed as a Syslog message will be transferred to this Relationship without any attributes being added")
+        .build();
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("Any FlowFile that is successfully parsed as a Syslog message will be to this Relationship.")
+        .build();
+
+    private StrictSyslog5424Parser parser;
```

Comment:
Since this gets initialized in onTrigger which can be called by multiple different threads, it is generally a good idea to mark it as volatile

Reply:
done